packages/react-instantsearch-core/src/connectors/connectConfigureRelatedItems.ts

```diff
@@ -0,0 +1,220 @@
+import algoliasearchHelper, {
+  PlainSearchParameters,
+  SearchParameters,
+} from 'algoliasearch-helper';
+import createConnector, { ConnectedProps } from '../core/createConnector';
+import {
+  omit,
+  getObjectType,
+  removeEmptyKey,
+  removeEmptyArraysFromObject,
+} from '../core/utils';
+import {
+  refineValue,
+  getIndexId,
+  hasMultipleIndices,
+} from '../core/indexUtils';
+
+type Hit = any;
+
+export type MatchingPatterns = {
+  [attribute: string]: {
+    /**
+     * The score of the optional filter.
+     *
+     * @see https://www.algolia.com/doc/guides/managing-results/rules/merchandising-and-promoting/in-depth/optional-filters/
+     */
+    score: number;
+  };
+};
+
+interface ConfigureRelatedItemsProps {
+  /**
+   * The reference hit to extract the filters from.
+   */
+  hit: Hit;
+  /**
+   * The schema to create the optional filters.
+   * Each key represents an attribute from the hit.
+   */
+  matchingPatterns: MatchingPatterns;
+  /**
+   * Function called to transform the search parameters generated.
+   */
+  transformSearchParameters?(
+    searchParameters: SearchParameters
+  ): PlainSearchParameters;
+}
+
+function createOptionalFilter({
+  attributeName,
+  attributeValue,
+  attributeScore,
+}) {
+  return `${attributeName}:${attributeValue}<score=${attributeScore || 1}>`;
+}
+
+const defaultProps: Partial<ConfigureRelatedItemsProps> = {
+  transformSearchParameters: x => ({ ...x }),
+};
+
+function getId(): string {
+  // We store the search state of this widget in `configure`.
+  return 'configure';
+}
+
+type InternalConfigureRelatedItemsProps = ConfigureRelatedItemsProps &
+  Required<typeof defaultProps>;
+
+function getSearchParametersFromProps(
+  props: ConnectedProps<InternalConfigureRelatedItemsProps>
+): PlainSearchParameters {
+  const optionalFilters = Object.keys(props.matchingPatterns).reduce<
+    Array<string | string[]>
+  >((acc, attributeName) => {
+    const attribute = props.matchingPatterns[attributeName];
```

Comment:
```suggestion
    const attributePattern = props.matchingPatterns[attributeName];
```

not sure if this makes more sense, but I was confused what this could be initially

Reply:
maybe also switch line 75 & 76

Reply:
Clearer indeded.